eras/conway/impl/src/Cardano/Ledger/Conway/Governance/Snapshots.hs

```diff
@@ -153,9 +150,7 @@ snapshotLookupId gId ProposalsSnapshot {psGovActionStates} =
 fromGovActionStateSeq ::
   StrictSeq (GovActionState era) ->
   ProposalsSnapshot era
-fromGovActionStateSeq (x :<| xs) =
-  snapshotInsertGovAction x (fromGovActionStateSeq xs)
```

Comment:
Isn't foldl' processing the list in a similar way like this recursive function:  the last one in the list would be the first one to be inserted?

Reply:
Yes, except, `snapshotInsertGovAction` appends elements at the end of the sequence, instead of at the beginning, which resulted in sequence being reversed every time it was deserialized :grin:

Reply:
Ah yeah, the original function looks more like the shape of `foldr`, which explains the difference in order.   Nice subtle bug :smile:

Reply:
Nice catch! :sweat_smile: